pkg/virt-controller/services/template.go

```diff
@@ -1447,6 +1448,7 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		vmiCPUArch = t.clusterConfig.GetClusterCPUArch()
 	}
 	memoryOverhead := GetMemoryOverhead(vmi, vmiCPUArch, t.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
+	metrics.SetVmiLaucherMemoryOverhead(vmi, memoryOverhead)
```

Comment:
Aren't we supposed to keep the metric bits under `./pkg/monitoring`?

Reply:
At this stage, only metrics definitions and more Prometheus-related concepts, for example, in this example we are not using labels nor setting the values directly. Everything is handled by the `SetVmiLaucherMemoryOverhead` function.

The alternative here would be a collector to fetch the virt-launcher requested memory minus the domain requested memory. Would that be accurate?

Reply:
> The alternative here would be a collector to fetch the virt-launcher requested memory minus the domain requested memory. Would that be accurate?

Unfortunately it wouldn't. For example, if both memory request and guest memory are explicitly set (e.g. request=2Gi, guest=2Gi. the pod is overcommitted, but you can't know the calculated infra overhead).

I tend to think that the guest's calculated overhead is important enough in order to expose it. We can do so either with a new status field, or even with an annotation. I think that by doing so we'd gain:
* Eliminate the addition of monitoring code into non `./pkg/monitoring` packages.
* Let users understand the full picture w.r.t. the VM's memory calculations by looking at the VM's manifest only (i.e. not depending on Prometheus metrics for that):
  * The requested memory. (exists today)
  * The guest memory. (exists today)
  * The calculated infra overhead.
  * (And hence ->) the overcommitment ratio.
    * (should we expose that as well via status/annotation?)

@fabiand @vladikr Do you have any thoughts regarding this?

Reply:
maybe, not to block following work depending on this metric, we can move forward with this implementation, and I can open a follow-up issue/PR to discuss the alternatives

Reply:
Sure, sounds good :+1: 
please cc me in that issue :)